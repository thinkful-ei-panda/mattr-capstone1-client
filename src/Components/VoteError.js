import React, { Component } from "react";
import { Link } from "react-router-dom";
import Nav from "./Nav";
import './Vote/Vote.css'
export default class VoteConfirmation extends Component {
  render() {
    return (
      <div>
        <Nav />
        <h3>
          You have successfully voted.
        </h3>
        <Link to="/Election" className="vote-link">
          Go Back
        </Link>
      </div>
    );
  }
}
