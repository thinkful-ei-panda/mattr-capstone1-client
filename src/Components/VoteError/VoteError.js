import React, { Component } from "react";
import { Link } from "react-router-dom";
import Nav from "../Nav/Nav";
import '../Vote/Vote.css'
export default class VoteConfirmation extends Component {
  render() {
    return (
      <div>
        <Nav />
        <h3>
          You have already voted.You can vote twice.
        </h3>
        <Link to="/Election" className="vote-link">
          Go Back
        </Link>
      </div>
    );
  }
}
