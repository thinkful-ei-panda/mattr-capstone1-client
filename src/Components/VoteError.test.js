import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import VoteError from "./VoteError";

describe(`VoteError Component`, () => {
  describe(`Smoke test`, () => {
    it(`Renders without crashing`, () => {
      const div = document.createElement("div");
      ReactDOM.render(
        <BrowserRouter>
          <VoteError />
        </BrowserRouter>,
        div
      );
      ReactDOM.unmountComponentAtNode(div);
    });
  });

  describe(`Snapshot test`, () => {
    it(`Renders the UI as expected`, () => {
      const tree = renderer
        .create(
          <BrowserRouter>
            <VoteError />
          </BrowserRouter>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
