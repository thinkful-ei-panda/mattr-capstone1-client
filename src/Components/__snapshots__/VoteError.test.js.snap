// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VoteError Component Snapshot test Renders the UI as expected 1`] = `
<div>
  <div
    className="Nav"
  >
    <a
      href="/Home"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      href="/Election"
      onClick={[Function]}
    >
      Elections
    </a>
    <a
      href="/RegistrationForm"
      onClick={[Function]}
    >
       
      Register
       
    </a>
    <a
      href="/Login"
      onClick={[Function]}
    >
      Login
    </a>
  </div>
  <h3>
     
    You have already previously voted. You can not vote more than once.
  </h3>
  <a
    className="vote-link"
    href="/Election"
    onClick={[Function]}
  >
     
    Go Back
     
  </a>
</div>
`;
